src/sentry/middleware/auth.py

```diff
@@ -83,25 +88,3 @@ def process_exception(
             return expired(request, exception.user)
         else:
             return None
-
-
-class AuthenticationMiddleware(SessionAuthenticationMiddleware):
-    def process_request(self, request: HttpRequest) -> None:
-        request.user_from_signed_request = False
-
-        if request.path.startswith("/api/0/internal/rpc/"):
-            # Avoid doing RPC authentication when we're already
-            # in an RPC request.
-            request.user = AnonymousUser()
-            return
-
-        # If there is a valid signature on the request we override the
-        # user with the user contained within the signature.
-        user = process_signature(request)
-
-        if user is not None:
-            request.user = user
-            request.user_from_signed_request = True
-            return
```

Comment:
We no longer need this signature override behavior?

Reply:
Not in the middleware. We have the same logic applied to the two endpoints that need it via an Authentication adapter.